search/constraint.go

```diff
@@ -48,7 +48,8 @@ func filter(rg parquet.RowGroup, cs ...Constraint) ([]rowRange, error) {
 	}
 	rr := []rowRange{{from: int64(0), count: rg.NumRows()}}
 	for i := range cs {
-		srr, err := cs[i].filter(rg, i == 0, rr)
+		isPrimary := len(sc) > 0 && cs[i].path() == sc[0].Path()[0]
+		srr, err := cs[i].filter(rg, isPrimary, rr)
```

Comment:
@MichaHoffmann small bug on the constraint if we dont add in the constraint the "primary key".

Reply:
How so?

Reply:
if we only query something like:

```
	eq := Equal(schema.LabelToColumn("unique"), parquet.ValueOf("unique_0"))
	found := query(t, h.MinTime(), h.MaxTime(), lf, cf, eq)
	require.Len(t, found, totalMetricNames)
```

It assumes that the label "unique" is the primary key as `i=0`.

Reply:
Ah - that's why we always require the name label currently

Reply:
Yeah… we don’t wanna enforce that

Reply:
Okay! I think then we need to revisit that is primary logic - it's so we can use ascending logic which really only holds for primary sorting column.

Reply:
Ah yeah we can fix it - we just need to check that the primary sorting column. Appeared in matchers

Reply:
I think this fix works .. at least if worked on my unit test

Reply:
Yeah I think it works